src/main/java/org/osc/manager/ism/entities/SecurityGroupEntity.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.manager.ism.entities;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.osc.sdk.manager.element.ManagerSecurityGroupElement;
+
+@Entity
+@Table(name = "SECURITY_GROUP")
+public class SecurityGroupEntity implements ManagerSecurityGroupElement {
+
+	@Id
+	@GeneratedValue
+	@Column(name = "id")
+	private Long id;
+
+	@Column(name = "name", unique = true, nullable = false)
+	private String name;
+
+	@OneToMany(mappedBy = "securityGroup", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```

Comment:
do you need Cascade ALL? I had seen issues with this, for example when you delete the child even the parent gets deleted. can you check?

Reply:
I have seen the following behavior, CASCADE_ALL on parent entity==> the deletion operation on parent, deletes the child as well. The delete on child entity will be ignored.
CASCADE ALL on child entity ==> deletion operation on child, deletes the parent as well. The delete request on parent will be ignored.
May be it is good to check once.